src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -441,8 +466,28 @@ def iot_hub_create(cmd, client, hub_name, resource_group_name, location=None,
                                         sku=sku,
                                         properties=properties,
                                         tags=tags)
-
-    return client.iot_hub_resource.create_or_update(resource_group_name, hub_name, hub_description)
+    hub_description.identity = _build_identity(identities) if identities else None
+    if bool(identity_role) ^ bool(identity_scopes):
+        raise CLIError('At least one scope (--scopes) and one role (--role) required for system-assigned managed identity role assignment')
+
+    def identity_assignment(lro):
+        try:
+            from azure.cli.core.commands.arm import assign_identity
+            instance = lro.resource().as_dict()
+            identity = instance.get("identity")
+            if identity:
+                principal_id = identity.get("principal_id")
+                if principal_id:
+                    hub_description.identity.principal_id = principal_id
+                    for scope in identity_scopes:
+                        assign_identity(cmd.cli_ctx, lambda: hub_description, lambda hub: hub_description, identity_role=identity_role, identity_scope=scope)
+        except CloudError as e:
+            raise e
```

Comment:
May I ask what is the error message when a `CloudError` is thrown?

Reply:
It's just a direct log of the error that the service returns. In this instance, likely that the role either exists or that there was an error adding the role to the specific scope.

Since this could be assigning a role to multiple scopes, I figured it was best to let the user know exactly what error the service was throwing.

![image](https://user-images.githubusercontent.com/13545962/118777275-bd194d80-b83d-11eb-88f7-80ddb7df83fc.png)
![image](https://user-images.githubusercontent.com/13545962/118777317-c73b4c00-b83d-11eb-895b-aecd7cd9a049.png)

Reply:
OK